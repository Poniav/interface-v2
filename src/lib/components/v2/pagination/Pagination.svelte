<script lang="ts">
	import { buttonClasses } from '$lib/atoms/componentClasses';
	import Icon from '$lib/atoms/icons/Icon.svelte';
	import { chevronLeft, chevronRight } from 'svelte-awesome/icons';

	export let activePage: number;
	export let pageCount: number;
	export let handlePageChange: (page: number) => void;
	export let styleClass = '';

	$: pageArray = Array.from({ length: pageCount }, (_, i) => i + 1);
	$: isLastPage = activePage === pageArray.length;
	$: isFirstPage = activePage === 1;
</script>

<div class="btn-group {styleClass}">
	<button
		disabled={isFirstPage}
		data-testid="pagination-prev-button"
		class={buttonClasses.paginationButton}
		on:click={() => {
			if (activePage > 1) {
				handlePageChange(activePage - 1);
			}
		}}
	>
		<Icon
			data={chevronLeft}
			size={12}
			iconColorClass={isFirstPage ? 'icon-info' : 'icon-primary'}
		/>
	</button>
	{#each pageArray as page}
		{#if page > activePage - 4 && page < activePage + 4}
			<button
				disabled={page === activePage}
				class="{buttonClasses.paginationButton} {page === activePage ? 'text-primary' : ''}"
				on:click={() => handlePageChange(page)}
			>
				{page}
			</button>
		{/if}
	{/each}
	<button
		disabled={isLastPage}
		data-testid="pagination-next-button"
		class={buttonClasses.paginationButton}
		on:click={() => {
			if (activePage < pageArray.length) {
				handlePageChange(activePage + 1);
			}
		}}
	>
		<Icon
			data={chevronRight}
			size={12}
			iconColorClass={isLastPage ? 'icon-info' : 'icon-primary'}
		/>
	</button>
</div>
