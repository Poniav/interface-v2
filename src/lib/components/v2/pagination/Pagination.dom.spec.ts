import { render, fireEvent } from '@testing-library/svelte';
import Pagination from './Pagination.svelte'; // Adjust with the correct import path
import { describe, it, expect } from 'vitest';

describe('Pagination', () => {
	const handlePageChangeMock = vi.fn();
	it('renders correctly', async () => {
		const { getByText, container } = render(Pagination, {
			props: {
				activePage: 1,
				pageCount: 5,
				handlePageChange: () => {},
				styleClass: 'custom-class'
			}
		});

		expect(getByText('1')).toBeTruthy();

		expect(container).toMatchSnapshot();
	});

	it('disables the previous button on the first page', async () => {
		const { getByTestId } = render(Pagination, {
			props: {
				activePage: 1,
				pageCount: 5,
				handlePageChange: handlePageChangeMock
			}
		});

		const prevButton = getByTestId('pagination-prev-button');
		await fireEvent.click(prevButton);

		expect(handlePageChangeMock).toHaveBeenCalledTimes(0);
	});

	it('disables the next button on the last page', async () => {
		const { getByTestId } = render(Pagination, {
			props: {
				activePage: 5,
				pageCount: 5,
				handlePageChange: handlePageChangeMock
			}
		});

		const nextButton = getByTestId('pagination-next-button');
		await fireEvent.click(nextButton);
		expect(handlePageChangeMock).toHaveBeenCalledTimes(0);
	});

	it('enables previous and next buttons on middle pages & calls handlePageChange with the correct page when a page button is clicked', async () => {
		const { getByTestId } = render(Pagination, {
			props: {
				activePage: 3,
				pageCount: 5,
				handlePageChange: handlePageChangeMock
			}
		});

		const prevButton = getByTestId('pagination-prev-button');
		const nextButton = getByTestId('pagination-next-button');
		await fireEvent.click(nextButton);
		expect(handlePageChangeMock).toHaveBeenCalledWith(4);
		await fireEvent.click(prevButton);
		expect(handlePageChangeMock).toHaveBeenCalledWith(4);
		expect(handlePageChangeMock).toHaveBeenCalledTimes(2);
	});

	it('shows the correct range of page numbers around the active page', () => {
		const { getByText } = render(Pagination, {
			props: {
				activePage: 5,
				pageCount: 10,
				handlePageChange: handlePageChangeMock
			}
		});

		// Test for the presence of pages near the active page
		expect(getByText('4').innerHTML).toBe('4');
		expect(getByText('5').innerHTML).toBe('5');
		expect(getByText('5').className).contain('text-primary');
		expect(getByText('6').innerHTML).toBe('6');
	});
});
