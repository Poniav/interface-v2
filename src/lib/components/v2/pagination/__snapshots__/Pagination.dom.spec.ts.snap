// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > renders correctly 1`] = `
<body>
  <div>
    <div
      class="btn-group custom-class"
    >
      <button
        class="btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed"
        data-testid="pagination-prev-button"
        disabled=""
      >
        <svg
          aria-label=""
          class="fa-icon icon-info s-T5YyQewRBd7n"
          data-testid="icon"
          height="16"
          role="presentation"
          style="width:12px; height:12px; cursor:pointer;  "
          version="1.1"
          viewBox="0 0 1280 1792"
          width="11.428571428571429"
        >
          <path
            d="M1171 301l-531 531 531 531q19 19 19 45t-19 45l-166 166q-19 19-45 19t-45-19l-742-742q-19-19-19-45t19-45l742-742q19-19 45-19t45 19l166 166q19 19 19 45t-19 45z"
          />
          
           
           
          
          
        </svg>
        <!--&lt;Svg&gt;-->
        <!--&lt;Icon&gt;-->
        <!--&lt;Icon&gt;-->
      </button>
       
      <button
        class="btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed text-primary"
        disabled=""
      >
        1
      </button>
      
      <button
        class="btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed "
      >
        2
      </button>
      
      <button
        class="btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed "
      >
        3
      </button>
      
      <button
        class="btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed "
      >
        4
      </button>
      
      
       
      <button
        class="btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed"
        data-testid="pagination-next-button"
      >
        <svg
          aria-label=""
          class="fa-icon icon-primary s-T5YyQewRBd7n"
          data-testid="icon"
          height="16"
          role="presentation"
          style="width:12px; height:12px; cursor:pointer;  "
          version="1.1"
          viewBox="0 0 1280 1792"
          width="11.428571428571429"
        >
          <path
            d="M1107 877l-742 742q-19 19-45 19t-45-19l-166-166q-19-19-19-45t19-45l531-531-531-531q-19-19-19-45t19-45l166-166q19-19 45-19t45 19l742 742q19 19 19 45t-19 45z"
          />
          
           
           
          
          
        </svg>
        <!--&lt;Svg&gt;-->
        <!--&lt;Icon&gt;-->
        <!--&lt;Icon&gt;-->
      </button>
    </div>
    <!--&lt;Pagination&gt;-->
  </div>
</body>
`;
